plugins/woocommerce/src/Admin/Features/Blueprint/SettingOptions.php

```diff
@@ -4,162 +4,89 @@
-			'sections' => array(),
-			'options'  => array(),
-
-		);
-
-		foreach ( $page->get_sections() as $id => $section ) {
-			$section_id                      = Util::camel_to_snake( strtolower( $section ) );
-			$info['sections'][ $section_id ] = array(
-				'label'       => $section,
-				'subsections' => array(),
-			);
-
-			$settings = $page->get_settings_for_section( $id );
-
-			// Get subsections.
-			$subsections = array_filter(
-				$settings,
-				function ( $setting ) {
-					return isset( $setting['type'] ) && 'title' === $setting['type'] && isset( $setting['title'] );
-				}
-			);
-
-			foreach ( $subsections as $subsection ) {
-				if ( ! isset( $subsection['id'] ) ) {
-					$subsection['id'] = Util::camel_to_snake( strtolower( $subsection['title'] ) );
-				}
-
-				$info['sections'][ $section_id ]['subsections'][ $subsection['id'] ] = array(
-					'label' => $subsection['title'],
-				);
+		foreach ( $settings as $setting ) {
+			// Skip if the setting type is not valid.
+			if ( in_array( $setting['type'], $this->ignore_setting_types, true ) || ! isset( $setting['id'] ) ) {
+				continue;
 			}
 
-			// Get options.
-			$info['sections'][ $section_id ]['options'] = $this->get_page_section_settings( $settings, $page->get_id(), $section_id );
-		}
-		return $info;
-	}
-
+			$key = is_array( $setting['option_key'] ) ? $setting['option_key'][0] : $setting['option_key'];
 
-	/**
-	 * Get settings for a specific page section.
-	 *
-	 * @param array  $settings The settings.
-	 * @param string $page The page ID.
-	 * @param string $section The section ID.
-	 * @return array
-	 */
-	private function get_page_section_settings( $settings, $page, $section = '' ) {
-		$current_title = '';
-		$data          = array();
-		foreach ( $settings as $setting ) {
-			if ( 'sectionend' === $setting['type'] || 'slotfill_placeholder' === $setting['type'] || ! isset( $setting['id'] ) ) {
+			// Skip if the option key is already in the page options.
+			if ( in_array( $key, $page_options, true ) ) {
 				continue;
 			}
 
-			if ( 'title' === $setting['type'] ) {
-				$current_title = Util::camel_to_snake( strtolower( $setting['title'] ) );
-			} else {
-				$location = $page . '.' . $section;
-				if ( $current_title ) {
-					$location .= '.' . $current_title;
-				}
-
-				$data[] = array(
-					'id'       => $setting['id'],
-					'value'    => $this->wp_get_option( $setting['id'], $setting['default'] ?? null ),
-					'title'    => $setting['title'] ?? $setting['desc'] ?? '',
-					'location' => $location,
-				);
-			}
+			$page_options[ $key ] = get_option( $key, $setting['default'] ?? null );
 		}
-		return $data;
+
+		return $page_options;
 	}
 
 	/**
-	 * Merge page info options.
+	 * Get a settings page by ID.
 	 *
-	 * @param array $page_info The page info.
+	 * @deprecated 9.9.0 This method is no longer used.
+	 * @return null
+	 */
+	public function get_page() {
+		wc_deprecated_function( __METHOD__, '9.9.0' );
+		return null;
+	}
```

Comment:
How come we are deprecating this and the `get_page_info`? given this feature hasn't been officially released yet?
I would say, just remove them, unless your aware of places where it is already being extended?

Reply:
I was thinking about removing them too but I marked them as deprecated since the class was introduced in 9.8.0. But your point is valid, it should be safe to remove them since the feature is not released yet and don't see any reason why it's being used by 3rd parties. I'll remove them directly.

Reply:
Removed them in https://github.com/woocommerce/woocommerce/pull/57624/commits/643b53eb6b1a70e4c4f35339e5df773285878325